osu.Game/Screens/Select/Leaderboards/BeatmapLeaderboard.cs

```diff
@@ -241,10 +164,53 @@ protected override LeaderboardScore CreateDrawableTopScore(ScoreInfo model) => n
             Action = () => ScoreSelected?.Invoke(model)
         };
 
+        private void subscribeToLocalScores(CancellationToken cancellationToken)
+        {
+            scoreSubscription?.Dispose();
+            scoreSubscription = null;
+
+            if (beatmapInfo == null)
+                return;
+
+            scoreSubscription = realm.RegisterForNotifications(r =>
+                r.All<ScoreInfo>().Filter($"{nameof(ScoreInfo.BeatmapInfo)}.{nameof(BeatmapInfo.ID)} == $0"
+                                          + $" AND {nameof(ScoreInfo.Ruleset)}.{nameof(RulesetInfo.ShortName)} == $1"
+                                          + $" AND {nameof(ScoreInfo.DeletePending)} == false"
+                    , beatmapInfo.ID, ruleset.Value.ShortName), localScoresChanged);
+
+            void localScoresChanged(IRealmCollection<ScoreInfo> sender, ChangeSet changes, Exception exception)
+            {
+                if (cancellationToken.IsCancellationRequested)
+                    return;
+
+                var scores = sender.AsEnumerable();
+
+                if (filterMods && !mods.Value.Any())
+                {
+                    // we need to filter out all scores that have any mods to get all local nomod scores
+                    scores = scores.Where(s => !s.Mods.Any());
+                }
+                else if (filterMods)
+                {
+                    // otherwise find all the scores that have *any* of the currently selected mods (similar to how web applies mod filters)
+                    // we're creating and using a string list representation of selected mods so that it can be translated into the DB query itself
+                    var selectedMods = mods.Value.Select(m => m.Acronym);
+                    scores = scores.Where(s => s.Mods.Any(m => selectedMods.Contains(m.Acronym)));
+                }
+
+                scores = scores.Detach();
+
+                scoreManager.OrderByTotalScoreAsync(scores.ToArray(), cancellationToken)
+                            .ContinueWith(ordered =>
```

Comment:
This looks like it needs to be scheduled. Reasoning being is that this is async and the implementation of `SetScores` is doing update-bound tasks (`Show()`/`Hide()` at least).

Reply:
It's already safe to some level due to the synchronisation context, but will add a schedule to localise execution I guess.

Reply:
Actually it isn't due to the use of `ContinueWith`, and not `await`+`ConfigureAwait(true)`